### PR TITLE
Add eslint-plugin-mocha, prettier

### PR DESCRIPTION
**Pull request by mroderick**
This PR adds two eslint plugins:

* eslint-plugin-mocha
* prettier (and eslint-config-prettier, eslint-plugin-prettier)

#### How to verify

1. Check out this branch
1. `npm link`
1. In a consuming project:
    1. `npm install eslint-plugin-mocha prettier eslint-config-prettier eslint-plugin-prettier`
    1. `npm link @studio/eslint-config`
    1. `npm run lint`
1. Observe the new lint violations

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,7 +71,6 @@ module.exports = {
       }
     }],
     'linebreak-style': [2, 'unix'],
-    'max-len': [2, 80, 4],
```
Comment (mantoni) [inline index.js:58]: Why removed? I don't see this covered by prettier.

**Review by mantoni — COMMENTED** [review]
Can you also take care of #1 as part of this change?

File: index.js (hunk 1 above)
Comment (mroderick) [reply index.js:58]: It is covered with https://prettier.io/docs/en/options.html#print-width, which uses 80 characters as default.

However, there's some conflict between how prettier formats text and what ESLint approves of:

> Usually, Prettier takes care of following a maximum line length automatically. However, there are cases where Prettier can’t do anything, such as for long strings, regular expressions and comments. Those need to be split up by a human.
>
> If you’d like to enforce an even stricter maximum line length policy than Prettier can provide automatically, you can enable this rule. Just remember to keep `max-len`’s options and Prettier’s `printWidth` option in sync.
>
> Keep in mind that you might have to refactor code slightly if Prettier formats lines in a way that the `max-len` rule does not approve of.

https://github.com/prettier/eslint-config-prettier#max-len

Comment (mantoni) [reply index.js:58]: That's fair. I'm fine with exceptional cases going over the limit.

**Review by mantoni — APPROVED** [review]
👍

**Pull request merged**